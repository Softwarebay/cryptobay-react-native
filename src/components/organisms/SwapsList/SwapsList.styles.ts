import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  swapsList: {
    width: '100%',
  }
})