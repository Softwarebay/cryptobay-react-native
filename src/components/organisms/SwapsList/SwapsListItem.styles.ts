import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  swapListItem: {
    marginBottom: 24,
  },
  contentWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  leftColumn: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  valueInDolar: {
    alignSelf: 'flex-end',
  },
});
