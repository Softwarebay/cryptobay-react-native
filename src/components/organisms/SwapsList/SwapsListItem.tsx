import React from 'react';
import { TouchableOpacity, View } from 'react-native';

import { Swap } from '../../../types/swap.type';
import { Badge, Caption, Icon, Icons, Label } from '../../atoms';

import { styles } from './SwapsListItem.styles';
import { SwapDetails } from './swapDetails.type';

type SwapsListItemProps = Swap & {
  tokenShortName: string;
  testID?: string;
  onPressDetails: (details: SwapDetails) => void;
};

export const SwapsListItem: React.FC<SwapsListItemProps> = ({
  tokenShortName,
  date,
  operation,
  status,
  value,
  valueInDolar,
  from,
  to,
  testID,
  onPressDetails,
}: SwapsListItemProps) => {
  const iconColorByStatus = status === 'Confirmed' ? '#76E268' : '#EA3943';
  const rotateIconByOperation = operation === 'Received' && {
    transform: [{ rotate: '180deg' }],
  };
  const operationLabel = `${operation} ${tokenShortName}`;
  const badgeColorByStatus = status === 'Confirmed' ? 'green' : 'red';
  const valueLabel = `${value} ${tokenShortName}`;
  const dolarCaption = `$${valueInDolar}`;

  const swapDetails = {
    shortName: tokenShortName,
    operation: operationLabel,
    status: status,
    date: date,
    value: value,
    from: from,
    to: to,
  } as SwapDetails;

  const handleOnPressDetails = () => onPressDetails(swapDetails);

  return (
    <TouchableOpacity
      testID={testID}
      style={styles.swapListItem}
      onPress={handleOnPressDetails}>
      <Caption value={date} />

      <View style={styles.contentWrapper}>
        <View style={styles.leftColumn}>
          <Icon
            type={Icons.ARROW}
            width={40}
            height={40}
            color={iconColorByStatus}
            style={rotateIconByOperation}
          />

          <View>
            <Label value={operationLabel} />
            <Badge value={status} variant={badgeColorByStatus} />
          </View>
        </View>

        <View>
          <Label value={valueLabel} />
          <Caption value={dolarCaption} style={styles.valueInDolar} />
        </View>
      </View>
    </TouchableOpacity>
  );
};
