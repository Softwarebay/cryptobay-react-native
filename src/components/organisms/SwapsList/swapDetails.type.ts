export type SwapDetails = {
  shortName: string;
  operation: string;
  status: 'Confirmed' | 'Cancelled';
  date: string;
  value: number;
  from: string;
  to: string;
};
