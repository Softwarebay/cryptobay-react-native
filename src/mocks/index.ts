export { bnbSwapsHistory } from './data/bnbSwapsHistory.mock';
export { etheriumTockenMock } from './data/etheriumTocken.mock';
export { tokensMock } from './data/tokens.mock';
