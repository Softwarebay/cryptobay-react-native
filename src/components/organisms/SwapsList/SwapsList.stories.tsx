import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';
import { bnbSwapsHistory } from '../../../mocks';
import { SwapsList } from './SwapsList';

storiesOf('Organisms/SwapsList', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Swaps List', () => (
    <SwapsList
      tokenShortName={text('tokenShortName', 'BNB')}
      swapsHistory={bnbSwapsHistory}
    />
  ));
