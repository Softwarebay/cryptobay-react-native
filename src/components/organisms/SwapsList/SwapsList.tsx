import React from 'react';
import { View } from 'react-native';
import { FlatList } from 'react-native-gesture-handler';

import { Swap } from '../../../types/swap.type';

import { styles } from './SwapsList.styles';
import { SwapsListItem } from './SwapsListItem';

type SwapsListProps = {
  tokenShortName: string;
  swapsHistory: Swap[];
};

export const SwapsList: React.FC<SwapsListProps> = ({
  tokenShortName,
  swapsHistory,
}: SwapsListProps) => {
  return (
    <View style={styles.swapsList}>
      <FlatList
        data={swapsHistory}
        renderItem={({ item }) => (
          <SwapsListItem
            tokenShortName={tokenShortName}
            date={item.date}
            operation={item.operation}
            status={item.status}
            value={item.value}
            valueInDolar={item.valueInDolar}
            from={item.from}
            to={item.to}
          />
        )}
      />
    </View>
  );
};
