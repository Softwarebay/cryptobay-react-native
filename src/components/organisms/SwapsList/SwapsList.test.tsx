import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { bnbSwapsHistory } from '../../../mocks';

import { SwapsList } from './SwapsList';
import { SwapsListItem } from './SwapsListItem';

const mockOnPressDetails = jest.fn();

describe('SwapsList Organism', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <SwapsList
          tokenShortName="BNB"
          swapsHistory={bnbSwapsHistory}
          onPressDetails={mockOnPressDetails}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure swap list item navigate to item', () => {
    render(
      <SwapsListItem
        testID="swap-list-item"
        date="Mar 3 at 10:04am"
        operation="Received"
        status="Confirmed"
        value={0.04}
        valueInDolar={9.58799}
        from="0x3Dc6...DfCE"
        to="0x3Dc6...DfCE"
        tokenShortName="BNB"
        onPressDetails={mockOnPressDetails}
      />,
    );
    fireEvent.press(screen.getByTestId('swap-list-item'));
    expect(mockOnPressDetails).toHaveBeenCalledWith({
      shortName: 'BNB',
      operation: 'Received BNB',
      status: 'Confirmed',
      date: 'Mar 3 at 10:04am',
      value: 0.04,
      from: '0x3Dc6...DfCE',
      to: '0x3Dc6...DfCE',
    });
  });
});
