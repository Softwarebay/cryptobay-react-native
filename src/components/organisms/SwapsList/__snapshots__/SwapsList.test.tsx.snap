// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapsList Organism Component Snapshot 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <RCTScrollView
    collapsable={false}
    data={
      Array [
        Object {
          "date": "Mar 3 at 10:04am",
          "from": "0x3Dc6...DfCE",
          "id": 1,
          "operation": "Received",
          "status": "Confirmed",
          "to": "0x3Dc6...DfCE",
          "value": 0.04,
          "valueInDolar": 9.58799,
        },
        Object {
          "date": "Mar 3 at 10:04am",
          "from": "0x3Dc6...DfCE",
          "id": 2,
          "operation": "Sent",
          "status": "Cancelled",
          "to": "0x3Dc6...DfCE",
          "value": 2.35,
          "valueInDolar": 547.5265,
        },
        Object {
          "date": "Mar 3 at 10:04am",
          "from": "0x3Dc6...DfCE",
          "id": 3,
          "operation": "Received",
          "status": "Confirmed",
          "to": "0x3Dc6...DfCE",
          "value": 1.876,
          "valueInDolar": 436.11371,
        },
        Object {
          "date": "Mar 3 at 10:04am",
          "from": "0x3Dc6...DfCE",
          "id": 4,
          "operation": "Sent",
          "status": "Confirmed",
          "to": "0x3Dc6...DfCE",
          "value": 0.04,
          "valueInDolar": 9.58799,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
    waitFor={
      Array [
        Object {
          "current": null,
        },
        Object {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 24,
              "opacity": 1,
            }
          }
          testID="swap-list-item-1"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "lineHeight": 18,
                },
                Object {
                  "color": "#ABAFC4",
                },
                undefined,
              ]
            }
          >
            Mar 3 at 10:04am
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "rotate": "180deg",
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="40"
                  bbWidth="40"
                  fill="none"
                  focusable={false}
                  height={40}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                  testID="svg-icon-arrow"
                  vbHeight={24}
                  vbWidth={24}
                  width={40}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M11.5 18L11.5 6"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4285981288,
                          "type": 0,
                        }
                      }
                    />
                    <RNSVGPath
                      d="M7 10L11.5 6L16 10"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4285981288,
                          "type": 0,
                        }
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                      },
                      Object {
                        "fontSize": 16,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Received BNB
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                        "fontSize": 14,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#76E268",
                      },
                      undefined,
                    ]
                  }
                >
                  Confirmed
                </Text>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                  ]
                }
              >
                0.04 BNB
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "lineHeight": 18,
                    },
                    Object {
                      "color": "#ABAFC4",
                    },
                    Object {
                      "alignSelf": "flex-end",
                    },
                  ]
                }
              >
                $9.58799
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 24,
              "opacity": 1,
            }
          }
          testID="swap-list-item-2"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "lineHeight": 18,
                },
                Object {
                  "color": "#ABAFC4",
                },
                undefined,
              ]
            }
          >
            Mar 3 at 10:04am
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={false}
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="40"
                  bbWidth="40"
                  fill="none"
                  focusable={false}
                  height={40}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                  testID="svg-icon-arrow"
                  vbHeight={24}
                  vbWidth={24}
                  width={40}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M11.5 18L11.5 6"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4293540163,
                          "type": 0,
                        }
                      }
                    />
                    <RNSVGPath
                      d="M7 10L11.5 6L16 10"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4293540163,
                          "type": 0,
                        }
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                      },
                      Object {
                        "fontSize": 16,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Sent BNB
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                        "fontSize": 14,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#EA3943",
                      },
                      undefined,
                    ]
                  }
                >
                  Cancelled
                </Text>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                  ]
                }
              >
                2.35 BNB
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "lineHeight": 18,
                    },
                    Object {
                      "color": "#ABAFC4",
                    },
                    Object {
                      "alignSelf": "flex-end",
                    },
                  ]
                }
              >
                $547.5265
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 24,
              "opacity": 1,
            }
          }
          testID="swap-list-item-3"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "lineHeight": 18,
                },
                Object {
                  "color": "#ABAFC4",
                },
                undefined,
              ]
            }
          >
            Mar 3 at 10:04am
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "rotate": "180deg",
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="40"
                  bbWidth="40"
                  fill="none"
                  focusable={false}
                  height={40}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                  testID="svg-icon-arrow"
                  vbHeight={24}
                  vbWidth={24}
                  width={40}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M11.5 18L11.5 6"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4285981288,
                          "type": 0,
                        }
                      }
                    />
                    <RNSVGPath
                      d="M7 10L11.5 6L16 10"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4285981288,
                          "type": 0,
                        }
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                      },
                      Object {
                        "fontSize": 16,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Received BNB
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                        "fontSize": 14,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#76E268",
                      },
                      undefined,
                    ]
                  }
                >
                  Confirmed
                </Text>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                  ]
                }
              >
                1.876 BNB
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "lineHeight": 18,
                    },
                    Object {
                      "color": "#ABAFC4",
                    },
                    Object {
                      "alignSelf": "flex-end",
                    },
                  ]
                }
              >
                $436.11371
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 24,
              "opacity": 1,
            }
          }
          testID="swap-list-item-4"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "lineHeight": 18,
                },
                Object {
                  "color": "#ABAFC4",
                },
                undefined,
              ]
            }
          >
            Mar 3 at 10:04am
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={false}
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="40"
                  bbWidth="40"
                  fill="none"
                  focusable={false}
                  height={40}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 40,
                        "width": 40,
                      },
                    ]
                  }
                  testID="svg-icon-arrow"
                  vbHeight={24}
                  vbWidth={24}
                  width={40}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M11.5 18L11.5 6"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4285981288,
                          "type": 0,
                        }
                      }
                    />
                    <RNSVGPath
                      d="M7 10L11.5 6L16 10"
                      propList={
                        Array [
                          "stroke",
                        ]
                      }
                      stroke={
                        Object {
                          "payload": 4285981288,
                          "type": 0,
                        }
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                      },
                      Object {
                        "fontSize": 16,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Sent BNB
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                        "fontSize": 14,
                        "fontWeight": "600",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#76E268",
                      },
                      undefined,
                    ]
                  }
                >
                  Confirmed
                </Text>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                  ]
                }
              >
                0.04 BNB
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "lineHeight": 18,
                    },
                    Object {
                      "color": "#ABAFC4",
                    },
                    Object {
                      "alignSelf": "flex-end",
                    },
                  ]
                }
              >
                $9.58799
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
